### PR TITLE
add function generateThumbnail

### PR DESCRIPTION
**Pull request by mixcas**
generate thumbnails when an image is uploaded

### CODE REVIEW

**Comment by varffs** [comment]
I think we should add quite a lot of sizes me might want at this stage rather than later as we dont want to have to work out how to regenerate image sizes on already uploaded images. Better to have too many sizes than not enough

**Comment by varffs** [comment]
How do we code review stuff like this? I can't see any way to actually test this code (how are you writing it even?)

**Comment by varffs** [comment]
It looks fine tho except the sizes thing

**Comment by mixcas** [comment]
@varffs i hate it too cuz i'm working this and deploying it, testing it and realtime and i hate it. I couldn't make it work locally. I'm not sure how to but there must be a way.

I'll appreciate suggestions for the sizes.

**Comment by mikeyrayvon** [comment]
i suggest we derive sizes from device widths. 

375, 768, 1024 are iphone, ipad and ipad pro. retina so 750, 1536, 2048.

and 300 works for admin thumb and grid view on iphone.

so our sizes array could be `[300, 750, 1536, 2048]`

what say you?

**Comment by varffs** [comment]
I think we should ad a mid size like 1024 or something like that and then it looks good. We should probably also add a really small 50 or 100 or something that might be useful. Are we building in the ability to square crop at all?

**Comment by mixcas** [comment]
Yeah, a 300x300 hard crop would be good.

I'm gonna do all those changes. I also wanna clean the code a bit.

**Comment by mixcas** [comment]
Added the suggested sizes and a better declaration. For each of the suggested sizes we generate resized images of given sizes and squared cropped images of width and height of given sizes.

Ex, `350`, generates a 350px max width image and a 350x350px image.

Updated readme

File: functions/test/generateThumbnail.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  resourceState:'exists',
+  selfLink:'https://www.googleapis.com/storage/v1/b/igv-andamiaje-co.appspot.com/o/uploads%2Ftumblr_mmqwshEaAG1qg1dqyo1_1280.jpg',
+  size:'252393',
+  storageClass:'STANDARD',
+  timeCreated:'2018-02-01T20:39:15.477Z',
+  timeStorageClassUpdated:'2018-02-01T20:39:15.477Z',
+  updated:'2018-02-01T20:39:15.477Z'
+};
+
+generateThumbnail(data);
```
Comment (varffs) [inline functions/test/generateThumbnail.js:26]: should this test run some kinda of assertation on the return value?

**Comment by varffs** [comment]
Read the code it looks good. Ran the test it worked.

File: functions/test/generateThumbnail.js (hunk 1 above)
Comment (mixcas) [reply functions/test/generateThumbnail.js:26]: Idk what it returns. Would be good to check

**Pull request merged**